appservice/src/deploy/localGitDeploy.ts

```diff
@@ -32,30 +33,65 @@ export async function localGitDeploy(client: SiteClient, fsPath: string): Promis
             await ext.ui.showWarningMessage(message, { modal: true }, deployAnyway, DialogResponses.cancel);
         }
         await verifyNoRunFromPackageSetting(client);
-        // tslint:disable-next-line:no-floating-promises
-        localGit.push(remote, 'HEAD:master');
+
+        // wrap two async actions in a promise because both need to run and finish simaltaneously
+        await new Promise((resolve: () => void, reject: (err: IParsedError) => void): void => {
+            ext.outputChannel.appendLine(formatDeployLog(client, (localize('localGitDeploy', `Deploying Local Git repository to "${client.fullName}"...`))));
+            localGit.push(remote, 'HEAD:master').catch(async (error) => {
+                tokenSource.cancel();
+                // due to timing issues, output here when we know we need to cancel
+                ext.outputChannel.appendLine(formatDeployLog(client, localize('cancelledDeployment', 'Cancelled deployment.')));
+                // tslint:disable-next-line:no-unsafe-any
+                const parsedError: IParsedError = parseAndRemoveCredentialsFromError(error);
+
+                // if the push fails, check if it's due to diverging history to offer force push
+                if (parsedError.message.indexOf('Updates were rejected because the remote contains work that you do') >= 0) {
+                    await forcePush();
```

Comment:
I think errors from forcePush are being ignored right now

Reply:
Does it not throw the error, and then get parsed in the catch? Or do you mean errors from the force push itself? (since I don't await that)

Reply:
You're inside of `await new Promise(...` so the only errors that are tracked from any of this code are the ones passed in to `reject`